### PR TITLE
Add check script

### PR DESCRIPTION
**Pull request by pvdrz**
# Description of change

This PR adds a `check.sh` script that could be run by contributors before opening a pull request to be (almost) sure that their changes will pass our CI process.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

The script was tested over the `coordicide` branch and ran successfully.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: check.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    status_code=$?
+    if [[ $status_code -eq 0 ]] ; then
+        echo $cmd was succesful!
+    else
+        echo $cmd failed!
+    fi
+    return $status_code
+}
+
+#
```
Comment (Adam-Gleave) [inline check.sh]: Should this be a comment? ("If no arguments", etc)

**Review by Adam-Gleave — COMMENTED** [review]
Do we definitely want this in the root? (It doesn't bother me, but might be worth discussing).

It also may be worth renaming to something like `checks.sh` or `all_checks.sh`, to differentiate it from `cargo check` or the check workflow specifically.

**Comment by pvdrz** [comment]
> Do we definitely want this in the root? (It doesn't bother me, but might be worth discussing).
> 
> It also may be worth renaming to something like `checks.sh` or `all_checks.sh`, to differentiate it from `cargo check` or the check workflow specifically.

I like `all_checks.sh` :)

File: check.sh (hunk 1 above)
Comment (pvdrz) [reply check.sh]: Indeed

**Comment by thibault-martinez** [comment]
> Do we definitely want this in the root? (It doesn't bother me, but might be worth discussing).

No we don't :trollface:

**Comment by pvdrz** [comment]
@thibault-martinez where would you put it?

**Comment by thibault-martinez** [comment]
> @thibault-martinez where would you put it?

Maybe simply a `scripts` folder at the root ?

**Comment by pvdrz** [comment]
SGTM

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**